timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -457,6 +454,10 @@ def predict(
             predictions = self.target_scaler.inverse_transform(predictions)
         return predictions
 
+    def get_forecast_horizon_index(self, data: TimeSeriesDataFrame) -> pd.MultiIndex:
```

Comment:
design nit: Is there any reason left for `get_forecast_horizon_index_ts_dataframe` to exist in `utils`? If so, why are we having the models depend on it through their superclass?

Reply:
We have showed this method to the users in the [in-depth tutorial](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#time-varying-covariates) and I have seen a few examples of customers using it in their code. The functionality is non-trivial to implement on your own and I don't want to break this without warning. How about the following plan?
1. We add the deprecation decorator to the method in `utils.forecast` and hard-remove it in v2.0
2. We add a method `get_forecast_horizon_index` to `TimeSeriesPredictor` so that users still have this functionality (see #4792)

Reply:
Sounds good to me.

Reply:
Will do the deprecation in a separate PR. For now just added a TODO